src/main/java/com/salesforce/storm/spout/dynamic/kafka/Consumer.java

```diff
@@ -329,6 +363,48 @@ public Record nextRecord() {
     }
 
     /**
+     * Reports kafka consumer statistics at an interval.
+     */
+    private void reportStatus() {
+        // If we've reported status more recently than 30 seconds we skip over this.
+        if (lastReportedStatusMillis != null && TimeUnit.MILLISECONDS.toSeconds(clock.millis() - lastReportedStatusMillis) < 30) {
+            return;
+        }
+
+        final Map<TopicPartition, Long> topicPartitions = getKafkaConsumer().endOffsets(
+            getKafkaConsumer().assignment()
+        );
+
+        // Capture the ending offset for this topic/partition on the kafka topic
+        for (final Map.Entry<TopicPartition, Long> topicPartition : topicPartitions.entrySet()) {
+            metricsRecorder.assignValue(
+                getClass(),
+                "endOffset.topic." + topicPartition.getKey().topic() + ".partition." + topicPartition.getKey().partition(),
+                topicPartition.getValue()
+            );
+        }
+
+        // Capture the difference, or the "lag" for this topic/partition on the kafka topic
+        for (final ConsumerPartition consumerPartition : getCurrentState().getConsumerPartitions()) {
+            final TopicPartition topicPartition = new TopicPartition(consumerPartition.namespace(), consumerPartition.partition());
+
+            if (!topicPartitions.containsKey(topicPartition)) {
+                continue;
+            }
+
+            final Long endOffset = topicPartitions.get(topicPartition);
+
+            metricsRecorder.assignValue(
+                getClass(),
+                "lag.topic." + consumerPartition.namespace() + ".partition." + consumerPartition.partition(),
```

Comment:
@Crim Another thing I've been thinking about lately... should we consider adding an enum of metric names and String.format() these, or even change the metrics recorder to take a series of values for use in String.format() similar to how the logger does? One of the big advantages of having an enum, or a class of Strings even will be that we can generate docs for supported metrics really easily, just like we do configuration values.

Reply:
Might be an interesting idea.  Perhaps part of a separate PR?